book: add rust snippet for NIP-17

rodant opened the pull request:
### Description

This PR adds  an Rust snippet for NIP-17 (private direct messages) to the book.

### Checklist

* [x] I followed the [contribution guidelines](https://github.com/rust-nostr/nostr/blob/master/CONTRIBUTING.md)
* [x] I ran `just precommit` or `just check` before committing
* [ ] I updated the [CHANGELOG](https://github.com/rust-nostr/nostr/blob/master/CHANGELOG.md) (if applicable)

yukibtc commented:
Thanks! I fixed some issues, moved it to `nostr-sdk` section and cherry-picked at 8544912be9fee667c457105f203640bd03d2c22d

rodant commented:
> Thanks! I fixed some issues, moved it to `nostr-sdk` section and cherry-picked at 8544912be9fee667c457105f203640bd03d2c22d

Ok, great, your changes make sense.
